hoomd/md/pytest/test_rigid.py

```diff
@@ -453,3 +453,152 @@ def test_rigid_dof(
+    "com_velocity",
+    [(0, 0, 0), (1, 2, 3)],
+)
+def test_velocity_constituents(
+    com_velocity,
+    simulation_factory,
+    one_particle_snapshot_factory,
+    valid_body_definition,
+):
+    """Test velocity of constituents updates correctly with constant torque."""
+    initial_snapshot = one_particle_snapshot_factory(particle_types=["A", "B"])
+    if initial_snapshot.communicator.rank == 0:
+        initial_snapshot.particles.moment_inertia[:] = [(1, 1, 1)]
+        initial_snapshot.particles.velocity[:] = [com_velocity]
+
+    sim = simulation_factory(initial_snapshot)
+    sim.seed = 5
+
+    rigid = md.constrain.Rigid()
+    rigid.body["A"] = valid_body_definition
+    rigid.create_bodies(sim.state)
+
+    constvol = md.methods.ConstantVolume(filter=hoomd.filter.Rigid())
+    integrator = md.Integrator(
+        dt=0.005,
+        methods=[constvol],
+        integrate_rotational_dof=True,
+    )
+    integrator.rigid = rigid
+    sim.operations += integrator
+    sim.run(1)
+
+    new_snapshot = sim.state.get_snapshot()
+    expected_velocity = [com_velocity] * 5
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
+
+    constant_torque = hoomd.md.force.Constant(filter=hoomd.filter.All())
+    constant_torque.constant_torque["A"] = (0, 0, 1)
+    integrator.forces.append(constant_torque)
+
+    sim.run(1)
+    # compare the resulting tangential velocities with expected values
+    expected_tangential_velocity = [
+        [0, 0, 0],
+        [-6.25e-08, 5.00e-03, 0.00e00],
+        [6.25e-08, -5.00e-03, 0.00e00],
+        [5.00e-03, 6.25e-08, 0.00e00],
+        [-5.00e-03, -6.25e-08, 0.00e00],
+    ]
+    expected_velocity = np.add(expected_tangential_velocity, com_velocity)
+    new_snapshot = sim.state.get_snapshot()
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
+
+
+@pytest.mark.parametrize(
+    "omega",
+    [
+        [10, 0, 0],
+        [0, 10, 0],
+        [0, 0, 10],
+        [3, 4, 10],
+    ],
+)
+@pytest.mark.parametrize(
+    "body_definition,moment_of_inertia,mass",
+    [
+        (
+            {
+                "constituent_types": ["B", "B", "B", "B"],
+                "positions": [
+                    [1, 0, -1 / (2 ** (1.0 / 2.0))],
+                    [-1, 0, -1 / (2 ** (1.0 / 2.0))],
+                    [0, -1, 1 / (2 ** (1.0 / 2.0))],
+                    [0, 1, 1 / (2 ** (1.0 / 2.0))],
+                ],
+                "orientations": [(1.0, 0.0, 0.0, 0.0)] * 4,
+            },
+            [1, 1, 1],
+            [1],
+        ),
+        (
+            {
+                "constituent_types": ["B", "B"],
+                "positions": [[-1.2, 0, 0], [1.2, 0, 0]],
+                "orientations": [[1, 0, 0, 0], [1, 0, 0, 0]],
+            },
+            [0.0, 2.88, 2.88],
+            [2],
+        ),
+    ],
+    ids=["simple-inertia-body", "dimer-body"],
+)
+@pytest.mark.parametrize(
+    "com_velocity",
+    [(0, 0, 0), (1, 2, 3)],
+)
```

Comment:
For readability, probably move this to above `omega` so that the last parametrization is the complicated one